src/test/java/cloud/fogbow/ras/core/processors/UnableToCheckStatusProcessorTest.java

```diff
@@ -1,313 +1,308 @@
     @Test
-    public void testRunProcessLocalOrderThrowsRuntimeException()
-			throws InterruptedException, FogbowException {
+    public void testRunProcessLocalOrderNotFailed() throws FogbowException, InterruptedException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.FULFILLED);
+        this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.unableToCheckStatus.getNext());
 
+        // exercise
+        this.processor.processUnableToCheckStatusOrder(order);
+
+        // verify
+        Assert.assertEquals(order, this.fulfilledOrderList.getNext());
+        Assert.assertNull(this.unableToCheckStatus.getNext());
+    }
+
+    // test case: When executing the thread in UnableToCheckStatusProcessor, if the
+    // instance state is still Failed after a successful request, the
+    // processUnableToCheckStatusOrder method should not change its state and it
+    // must remain in the list of failures.
+    @Test
+    public void testRunProcessLocalOrderWithInstanceFailed() throws FogbowException, InterruptedException {
         // set up
-		Order order = createOrder();
-		order.setOrderStateInTestMode(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST);
-		this.unableToCheckStatus.addItem(order);
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST);
+        this.unableToCheckStatus.addItem(order);
+        Assert.assertNull(this.fulfilledOrderList.getNext());
 
-        spyFailedProcessor();
+        OrderInstance orderInstance = new ComputeInstance(BaseUnitTests.FAKE_INSTANCE_ID);
+        orderInstance.setHasFailed();
+
+        Mockito.doReturn(orderInstance).when(this.cloudConnector).getInstance(Mockito.any(Order.class));
 
         // exercise
-        Mockito.doThrow(new RuntimeException()).when(this.unableToCheckStatusProcessor)
-                .processUnableToCheckStatusOrder(order);
+        this.thread = new Thread(this.processor);
+        this.thread.start();
+        Thread.sleep(BaseUnitTests.DEFAULT_SLEEP_TIME);
+
+        // verify
+        Assert.assertNotNull(this.unableToCheckStatus.getNext());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
+    }
 
-        this.thread = new Thread(this.unableToCheckStatusProcessor);
+    // test case: When executing the thread in the UnableToCheckStatusProcessor, if
+    // the instance is back to the Ready state, the processUnableToCheckStatusOrder
+    // method must change OrderState from UnableToCheckStatus to Fulfilled and the
+    // order must be removed from the unableToCheckStatus list and put in the
+    // fulfilled list.
+    @Test
+    public void testRunProcessLocalOrderWithInstanceReady() throws InterruptedException, FogbowException {
```

Comment:
why you handle with two orders here? Isn't one enough?

Reply:
The OrderInstance used here is a generic instance type provided as a response. But, I'll replace OrderInstance with ComputeInstance.